### ISSUE TITLE
Disable Http Server when node becomes unhealthy

### ISSUE THREAD

**Issue by ceyonur** [ava-labs/avalanchego#1277]
Nodes can get hammered by external services/requests. This can lead node to fall behind with the consensus and become unhealthy. If node is not behind a proper load-balancing these requests can prevent node to become healthy. We can add a mechanism to disable HTTP API server when node becomes unhealthy. Ideally this should disable relevant unhealthy chain API (instead disabling all of them). There should be a flag to enable/disable this behavior via Admin-API and CLI flag.

**Comment by danlaine** [comment]
Is it desirable to disable API calls when a chain becomes unhealthy? API calls can be useful in debugging the reason the node is unhealthy.

**Comment by ceyonur** [comment]
> Is it desirable to disable API calls when a chain becomes unhealthy? API calls can be useful in debugging the reason the node is unhealthy.

True. This won't be activated by default/ We've added a CLI flag, and also an Admin API method to disable/enable this feature.

**Comment by joshua-kim** [comment]
I don't know if we actually want to do this - the apis are by default not available publicly so it seems weird to load-shed requests at the node level. LB could also just implement load shed since that's the publicly exposed component to the user.
